xrtpy/response/temperature_response.py

```diff
@@ -0,0 +1,184 @@
+__all__ = [
+    "TemperatureResponseFundamental",
+]
+
+import pkg_resources
+import scipy.io
+import sunpy.time
+
+from astropy import units as u
+from astropy.constants import c, h
+from datetime import datetime
+from scipy import integrate, interpolate
+
+from xrtpy.response.channel import Channel, resolve_filter_name
+from xrtpy.response.effective_area import effective_area
+from xrtpy.util.time import epoch
+
+_c_Å_per_s = c.to(u.angstrom / u.second).value
+_h_eV_s = h.to(u.eV * u.s).value
+
+_CHIANTI_filename = pkg_resources.resource_filename(
+    "xrtpy", "response/data/XRT_emiss_model.default_CHIANTI.geny"
+)
+_CHIANTI_file = scipy.io.readsav(_CHIANTI_filename)
+_XRT_emiss_model_file = _CHIANTI_file["p0"]
+
+
+CHIANTI_file = {
+    "abundance_model": _XRT_emiss_model_file["ABUND_MODEL"][0],
+    "dens_model": _XRT_emiss_model_file["DENS_MODEL"][0],
+    "ioneq_model": _XRT_emiss_model_file["IONEQ_MODEL"][0],
+    "name": _XRT_emiss_model_file["NAME"][0],
+    "spectra": _XRT_emiss_model_file["SPEC"],
+    "spectra_units": _XRT_emiss_model_file["SPEC_UNITS"][0],
+    "temperature": _XRT_emiss_model_file["TEMP"][0],
+    "temp_units": _XRT_emiss_model_file["TEMP_UNITS"][0],
+    "tlength": _XRT_emiss_model_file["TLENGTH"][0],
+    "wlength": _XRT_emiss_model_file["WLENGTH"][0],
+    "wavelength": _XRT_emiss_model_file["WAVE"][0],
+    "wavelength_units": _XRT_emiss_model_file["WAVE_UNITS"][0],
+}
+
+
+class TemperatureResponseFundamental:
+    """Produce the temperature response for each XRT x-ray channel, assuming a spectral emission model."""
+
+    def __init__(self, filter_name, observation_date):
+        self._name = resolve_filter_name(filter_name)
+        self.observation_date = observation_date
+        self._channel = Channel(self.name)
+
+    @property
+    def name(self):
+        """Name of searched filter."""
+        return self._name
+
+    @property
+    def observation_date(self):
+        """Users date of observation."""
+        return self._observation_date
+
+    @observation_date.setter
+    def observation_date(self, date):
+        """Validating users requested observation date."""
+        astropy_time = sunpy.time.parse_time(date)  # Astropy time in utc
+        observation_date = astropy_time.datetime
+        if observation_date <= epoch:
+            raise ValueError(
+                "Invalid date: {:}.\n Date must be after September 22nd, 2006 21:36:00.".format(
+                    observation_date
```

Comment:
This could be changed to an [f-string](https://realpython.com/python-f-strings/#f-strings-a-new-and-improved-way-to-format-strings-in-python).